nipype/pipeline/plugins/multiproc.py

```diff
@@ -134,207 +114,190 @@ def __init__(self, plugin_args=None):
+                    continue
+                if num_subnodes > 1:
+                    submit = self._submit_mapnode(jobid)
+                    if not submit:
                         continue
-                    if num_subnodes > 1:
-                        submit = self._submit_mapnode(jobid)
-                        if not submit:
-                            continue
-
-                # change job status in appropriate queues
-                self.proc_done[jobid] = True
-                self.proc_pending[jobid] = True
-
-                free_memory_gb -= self.procs[jobid]._interface.estimated_memory_gb
-                free_processors -= self.procs[jobid]._interface.num_threads
-
-                # Send job to task manager and add to pending tasks
-                if self._status_callback:
-                    self._status_callback(self.procs[jobid], 'start')
-                if str2bool(self.procs[jobid].config['execution']['local_hash_check']):
-                    logger.debug('checking hash locally')
-                    try:
-                        hash_exists, _, _, _ = self.procs[
-                            jobid].hash_exists()
-                        logger.debug('Hash exists %s' % str(hash_exists))
-                        if (hash_exists and (self.procs[jobid].overwrite == False or
-                                             (self.procs[jobid].overwrite == None and
-                                              not self.procs[jobid]._interface.always_run))):
-                            self._task_finished_cb(jobid)
-                            self._remove_node_dirs()
-                            continue
-                    except Exception:
-                        etype, eval, etr = sys.exc_info()
-                        traceback = format_exception(etype, eval, etr)
-                        report_crash(self.procs[jobid], traceback=traceback)
-                        self._clean_queue(jobid, graph)
-                        self.proc_pending[jobid] = False
+
+            # Check requirements of this job
+            next_job_gb = min(self.procs[jobid].get_mem_gb(), self.memory_gb)
+            next_job_th = min(self.procs[jobid].get_n_procs(), self.processors)
+
+            # If node does not fit, skip at this moment
+            if next_job_th > free_processors or next_job_gb > free_memory_gb:
+                logger.debug('Cannot allocate job %d (%0.2fGB, %d threads).',
+                             jobid, next_job_gb, next_job_th)
+                continue
+
+            free_memory_gb -= next_job_gb
+            free_processors -= next_job_th
+            logger.info('Allocating %s ID=%d (%0.2fGB, %d threads). Free: %0.2fGB, %d threads.',
+                        self.procs[jobid]._id, jobid, next_job_gb, next_job_th,
+                        free_memory_gb, free_processors)
+
+            # change job status in appropriate queues
+            self.proc_done[jobid] = True
```

Comment:
Done? I see that's how this was before; just want to make sure this makes sense.

Reply:
This also struck me at first, but this is how it works :D. The name of this array is certainly confusing. It should be something like `proc_submitted` but that would change all plugins internally.

Reply:
the reason for this is the linear plugin, where it truly meant `proc_done`, but it was indeed short for proc_submission_done.